syncer/syncer.go

```diff
@@ -153,26 +166,45 @@ func (s *Syncer) IsSynced(ctx context.Context) bool {
 // Start starts the main sync loop that tries to sync data for every SyncInterval
 func (s *Syncer) Start(ctx context.Context) {
 	s.syncOnce.Do(func() {
-		if s.ticker.GetCurrentLayer().Uint32() <= 1 {
-			s.setSyncState(ctx, synced)
-		}
-		for {
-			select {
-			case <-s.shutdownCtx.Done():
-				s.logger.WithContext(ctx).Info("stopping sync to shutdown")
-				return
-			case <-s.syncTimer.C:
-				s.synchronize(ctx)
-			}
+		s.logger.WithContext(ctx).Info("Starting syncer loop")
+		err := s.tg.Go(s.loop)
+		if err != nil {
+			s.logger.With().Warning("taskgroup: Go returned an error",
+				log.Err(err))
+		} else {
+			s.setStarted()
 		}
 	})
 }
 
+func (s *Syncer) loop(ctx context.Context) error {
+	if s.ticker.GetCurrentLayer().Uint32() <= 1 {
+		s.setSyncState(ctx, synced)
+	}
+	for {
+		select {
+		case <-s.shutdownCtx.Done():
+			s.logger.WithContext(ctx).Info("stopping sync to shutdown")
+			return s.shutdownCtx.Err()
+		case <-s.syncTimer.C:
+			s.logger.WithContext(ctx).Debug("synchronize on tick")
+			s.synchronize(ctx)
+		}
+	}
+}
+
 // ForceSync manually starts a sync process outside the main sync loop. If the node is already running a sync process,
 // ForceSync will be ignored.
 func (s *Syncer) ForceSync(ctx context.Context) {
 	s.logger.WithContext(ctx).Debug("executing ForceSync")
-	go s.synchronize(ctx)
+	err := s.tg.Go(func(ctx context.Context) error {
```

Comment:
i think if it is possible to force a sync using a channel operation (instead of goroutine) then it will be safe to use errgroup in this case (https://github.com/spacemeshos/go-spacemesh/issues/2706)

Reply:
what do you mean by "using a channel operation"? currently it's called directly in cmd/sync/sync.go for testing purposes.

Reply:
i mean in the main loop we can add another channel, and when token is received from that channel goroutine to sync will be spawned. and in ForceSync we will send that token

Reply:
i see. thanks. i recall there was a ForceSync RPC before, but i can't find it anymore.  
do you actually prefer me using errgroup? i had assume that taskgroup is created for similar purpose as errgroup.

Reply:
i think there is a benefit in using errgroup, since it is simpler and doesn't require to return error from errgroup.Go.

Reply:
yes. much simpler. no need to create an error for errgroup.Wait() to return.
@dshulyak  please take another look. thanks

Reply:
it is great that we can use it. i am sort of unhappy with that taskgroup module, i expected it to be a bit simpler and easier to use. going to ditch it if errgroup can be used everywhere